rules/dynamically-load-components/rule.md

```diff
@@ -0,0 +1,50 @@
+---
+type: rule
+archivedreason: 
+title: Do you know when to use dynamic imports in Next.js?
+guid: b438c637-e0ee-4458-9180-43aa98fd9f0b
+uri: optimise-third-party-scripts
+created: 2023-07-14T02:13:28Z
+authors:
+- title: Harry Ross
+  url: https://www.ssw.com.au/people/harry-ross
+related:
+- use-nextjs
+- manage-bundle-size
+- improve-performance-with-lazy-loading-of-media-assets
+
+---
+
+Components with large imports loaded in on runtime may result in a much worse UX for users of your web app. Next.js can help with this by using [dynamic imports](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/import) to only load these large components when they are rendered on the page.
```

Comment:
I think we need to elaborate more on why it is a much worse UX for users here to show the pain point.

For the link, I think linking to NextJS Dynamic or lazy loading resources is better than linking to the current developer.mozilla one.

https://nextjs.org/docs/pages/building-your-application/optimizing/lazy-loading#nextdynamic

Reply:
Hm, I'm not sure how I would want to elaborate on how the UX is worse - maybe a PageSpeed Insights screenshot? Not sure how one would show loading improvements otherwise. 

I already link to that NextJS doc link below - and thought the Mozilla one would provide some background if required on how the underlying technology works - is it fine to repeat myself?